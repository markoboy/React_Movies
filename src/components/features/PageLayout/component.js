import { getRouteElement } from '@components/App/routes';
import SearchResults from '@components/features/SearchResults';
import SiteContainer from '@components/features/SiteContainer';
import React from 'react';
import { Switch } from 'react-router-dom';

export default function PageLayoutComponent({ route }) {
  return (
    <SiteContainer>
      <Switch>{route.routes.map(getRouteElement)}</Switch>

      <SearchResults />
    </SiteContainer>
  );
}
