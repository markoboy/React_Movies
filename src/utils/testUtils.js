/* eslint-disable import/no-extraneous-dependencies */
import defaultStore from '@store/index';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export function renderWithStore(
  ui,
  { store = defaultStore, route = '/', ...renderOptions } = {}
) {
  window.history.pushState({}, 'Test page', route);

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }

  return render(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

export function renderWithRouter(ui, { route = '/' } = {}) {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: BrowserRouter });
}
