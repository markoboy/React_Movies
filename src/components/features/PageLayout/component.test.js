import React from 'react';

import PageLayout from './component';
import { renderWithStore } from '@utils/testUtils';

jest.mock('@services/MovieServiceAPI', () => ({
  getAll: jest.fn(() => Promise.resolve([])),
  getOne: jest.fn(() => Promise.resolve({})),
  add: jest.fn(() => Promise.resolve({})),
  delete: jest.fn(() => Promise.resolve({})),
  update: jest.fn(() => Promise.resolve({})),
}));

describe('PageLayout Component', () => {
  afterAll(() => {
    jest.unmock('@services/MovieServiceAPI');
  });

  it('renders PageLayout Component with provided route', () => {
    const route = {
      routes: [
        {
          path: '/',
          exact: true,
          component: () => <p>Some component</p>,
        },
      ],
    };

    const { container } = renderWithStore(<PageLayout route={route} />);

    expect(container).toMatchSnapshot();
  });
});
