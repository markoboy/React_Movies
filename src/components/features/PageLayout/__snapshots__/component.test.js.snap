// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout Component renders PageLayout Component with provided route 1`] = `
<div>
  <main
    class="site-container__main flex flex--column"
  >
    <p>
      Some component
    </p>
    <div
      class="background--black flex flex--column flex--grow"
    >
      <div
        class="wrapper"
      >
        <ul
          class="result-filter flex"
        >
          <li
            class="result-filter__item result-filter__item--active"
          >
            All
          </li>
          <li
            class="result-filter__item"
          >
            Documentary
          </li>
          <li
            class="result-filter__item"
          >
            Comedy
          </li>
          <li
            class="result-filter__item"
          >
            Horror
          </li>
          <li
            class="result-filter__item"
          >
            Crime
          </li>
          <li
            class="result-filter__item result-filter__item--last"
          >
            <div
              class="result-filter__select-wrapper"
            >
              <label
                class="result-filter__sort-label"
                for="sort-options"
              >
                Sort by
              </label>
              <span
                class="result-filter__select-value"
              >
                Release Date
              </span>
              <select
                class="result-filter__select"
                id="sort-options"
                name="sort"
              >
                <option
                  value="release_date"
                >
                  Release Date
                </option>
                <option
                  value="vote_average"
                >
                  Rating
                </option>
              </select>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </main>
</div>
`;
